src/Login/index.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { loginUser } from '../actions/authentication';
+
+
+class Login extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            email: '',
+            password: '',
+            errors: {}
+        }
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleInputChange(e) {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        const user = {
+            email: this.state.email,
+            password: this.state.password,
+        }
+        this.props.loginUser(user);
+    }
+
+    componentDidMount() {
+        if(this.props.auth.isAuthenticated) {
+            this.props.history.push('/');
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if(nextProps.auth.isAuthenticated) {
+            this.props.history.push('/')
+        }
+        if(nextProps.errors) {
+            this.setState({
+                errors: nextProps.errors
+            });
+        }
+    }
+
+    render() {
+        const {errors} = this.state;
+        return(
+        <div className="container" style={{ marginTop: '50px', width: '700px'}}>
+            <h2 style={{marginBottom: '40px'}}>Login</h2>
+            <form onSubmit={ this.handleSubmit }>
+                <div className="form-group">
+                    <input
+                    type="email"
+                    placeholder="Email"
+                    className={classnames('form-control form-control-lg', {
+                        'is-invalid': errors.email
+                    })}
+                    name="email"
+                    onChange={ this.handleInputChange }
+                    value={ this.state.email }
+                    />
+                    {errors.email && (<div className="invalid-feedback">{errors.email}</div>)}
+                </div>
+                <div className="form-group">
+                    <input
+                    type="password"
+                    placeholder="Password"
+                    className={classnames('form-control form-control-lg', {
```

Comment:
move expression to variable and use here

Reply:
moved to variable